test/replication/gh-3637-misc-error-on-replica-auth-fail.result

```diff
@@ -49,6 +49,11 @@ vclock[0] = nil
 _ = test_run:wait_vclock('replica_auth', vclock)
 ---
 ...
+-- If we stop prematurely replica will not finish its start and
+-- will panic.
+_ = test_run:wait_log('replica_auth', 'entering the event loop')
```

Comment:
Wouldn't it be better to wait until `box.ctl.is_recovery_finished()` returns true?

Reply:
It is not clear can `box.cfg` yield or not after `box.ctl.is_recovery_finished()` is true. So change to next:

```diff
--- a/test/replication/gh-3637-misc-error-on-replica-auth-fail.test.lua
+++ b/test/replication/gh-3637-misc-error-on-replica-auth-fail.test.lua
@@ -24,6 +24,11 @@ vclock = test_run:get_vclock('default')
 vclock[0] = nil
 _ = test_run:wait_vclock('replica_auth', vclock)

+-- Wait server init script finish or server will panic on stop.
+test_run:switch('replica_auth')
+test_run:wait_cond(function() return _G.startup_finished == true end)
+
+test_run:switch('default')
--- a/test/replication/replica_auth.lua
+++ b/test/replication/replica_auth.lua
@@ -4,9 +4,12 @@ local USER_PASS = arg[1]
 local TIMEOUT = arg[2] and tonumber(arg[2]) or 0.1

 require('console').listen(os.getenv('ADMIN'))
+_G.startup_finished = false

 box.cfg({
     listen = os.getenv("LISTEN"),
     replication = USER_PASS .. "@" .. os.getenv("MASTER"),
     replication_timeout = TIMEOUT,
 })
+
+_G.startup_finished = true
```